builtin/provisioners/chef/resource_provisioner.go

```diff
@@ -370,18 +371,27 @@ func applyFn(ctx context.Context) error {
 }
 
 func validateFn(c *terraform.ResourceConfig) (ws []string, es []error) {
-	usePolicyFile, ok := c.Get("use_policyfile")
-	if !ok {
-		usePolicyFile = false
+	usePolicyFile := false
+	if usePolicyFileRaw, ok := c.Get("use_policyfile"); ok {
+		switch usePolicyFileRaw := usePolicyFileRaw.(type) {
+		case string:
+			if usePolicyFileBool, err := strconv.ParseBool(usePolicyFileRaw); err == nil {
+				usePolicyFile = usePolicyFileBool
+			} else {
+				es = append(es, errors.New("\"use_policyfile\": field is not a boolean value"))
+			}
+		case bool:
+			usePolicyFile = usePolicyFileRaw
+		}
```

Comment:
May I suggest a small rewrite of your `switch` statement to make it just a bit more idiomatic Go?

```go
switch usePolicyFileRaw := usePolicyFileRaw.(type) {
  case bool:
    usePolicyFile = usePolicyFileRaw
  case string:
    usePolicyFileBool, err := strconv.ParseBool(usePolicyFileRaw)
    if err != nil {
      return append(es, errors.New("\"use_policyfile\" must be a boolean"))
    }
    usePolicyFile = usePolicyFileBool
  default:
    return append(es, errors.New("\"use_policyfile\" must be a boolean")) 
}
```